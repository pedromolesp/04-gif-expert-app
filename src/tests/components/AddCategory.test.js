const { shallow } = require("enzyme")
const { AddCategory } = require("../../components/AddCategory")

const setCategories = jest.fn();
let wrapper = shallow(<AddCategory setCategories={setCategories} />);
const input = wrapper.find("input");
const value = "Hola mundo";
beforeEach(() => {

    wrapper = shallow(<AddCategory setCategories={setCategories} />);
    jest.clearAllMocks();
});
describe('Pruebas componente AddCategory', () => {

    test('debe mostrar el componente', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de cambiar la caja de texto', () => {
        input.simulate('change', { target: { value } });
        expect(wrapper.find('p').text().trim()).toBe(value)
    });
    test('no debe de postear la informaciÃ³n on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    });
    // test('debe de llamar el setCategorie y limpiar la caja de texto', () => {
    //     wrapper.find('input').simulate('change', { target: { value } });
    //     wrapper.find('form').simulate('submit', { preventDefault() { } });
    //     expect(setCategories).toHaveBeenCalled();
    //     expect(wrapper.find('input').prop('value')).toBe('');


    // })
    //ole


});
