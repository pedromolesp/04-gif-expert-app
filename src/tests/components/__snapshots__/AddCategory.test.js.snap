// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas componente AddCategory debe mostrar el componente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
